src/core/cloud.js

```diff
@@ -212,6 +212,21 @@ const putFile = (api, filePath, options) => {
 };
 exports.putFile = putFile;
 
+const deleteOnFailure = (api, options, createdId, exception) => {
+  if (createdId !== -1) {
+    return remove(`${api}/${createdId}`, null, options)
+      .then(cb => {
+        throw exception;
+      })
+      .catch(ex => {
+        logger.error('Error occured during trying to delete a resource even during handling the failure of operation');
```

Comment:
Let's add the `ex` here as an arg, i.e. 
```suggestion
        logger.error(`Error occurred on DELETE ${api}/${createdId} during cleanup`, ex);
```